src/algorithms/tracking/DD4hepBField.cc

```diff
@@ -10,18 +10,32 @@
 #include <Evaluator/DD4hepUnits.h>
 #include <Math/GenVector/DisplacementVector3D.h>
 #include <Eigen/Core>
+#include <limits>
 
 namespace eicrecon::BField {
 
   Acts::Result<Acts::Vector3> DD4hepBField::getField(const Acts::Vector3& position,
                                                      Acts::MagneticFieldProvider::Cache& /*cache*/) const
   {
-    dd4hep::Position pos(position[0]/10.0,position[1]/10.0,position[2]/10.0); // FIXME
-    auto fieldObj = m_det->field();
-
+    dd4hep::Position pos(
+      position[0] * (dd4hep::mm / Acts::UnitConstants::mm),
+      position[1] * (dd4hep::mm / Acts::UnitConstants::mm),
+      position[2] * (dd4hep::mm / Acts::UnitConstants::mm));
 
+    auto fieldObj = m_det->field();
     auto field = fieldObj.magneticField(pos) * (Acts::UnitConstants::T / dd4hep::tesla);
-    return Acts::Result<Acts::Vector3>::success({field.x(), field.y(),field.z()});
+
+    // FIXME Acts doesn't seem to like exact zero components
+    if (field.x() * field.y() * field.z() == 0) {
```

Comment:
That's the error I mentioned in mattermost a week or two ago. The EigenStepper starts to pass RK step sizes `h` that are NaN, as of v30 (but unclear it's related to a particular commit). I completely agree this should be reported. I haven't figured out a good reproducer though, and I've not been able to bisect it (because... yeah, Acts API changes prevent most bisecting). Just saying to the Acts devs "run the ePIC tracking" doesn't seem like a minimal example ;-)

Reply:
I wonder if one of the eigenstepper unit tests could be tested by just inputting a field value of 0 for one of the field components.